libminifi/test/archive-tests/MergeFileTests.cpp

```diff
@@ -752,3 +752,80 @@ TEST_CASE("MergeFileZip", "[mergefiletest5]") {
   }
   LogTestController::getInstance().reset();
 }
+
+TEST_CASE("MergeFileOnAttribute", "[mergefiletest5]") {
+  {
+    std::ofstream expectfileFirst(EXPECT_MERGE_CONTENT_FIRST, std::ios::binary);
+    std::ofstream expectfileSecond(EXPECT_MERGE_CONTENT_SECOND, std::ios::binary);
+
+    // Create and write to the test file
+    for (int i = 0; i < 6; i++) {
+      std::ofstream{std::string(FLOW_FILE) + "." + std::to_string(i) + ".txt", std::ios::binary} << std::to_string(i);
```

Comment:
Is there any advantage of performing this in binary mode?

Reply:
we should always read/write files in binary mode, unless we are up for some surprises, e.g. in non-binary mode on windows every '\n' character is prepended with a '\r'

Reply:
Well, I think this tests should not hide those differences. But we do not have whitespaces here at all, so the difference should not be relevant.